projects/igniteui-angular/src/lib/action-strip/grid-actions/grid-editing-actions.component.ts

```diff
@@ -147,7 +147,7 @@ export class IgxGridEditingActionsComponent extends IgxGridActionsBaseDirective
         const context = this.strip.context;
         const grid = context.grid;
         grid.deleteRow(context.key);
-
+        this.grid.cdr.detectChanges();
```

Comment:
Okay, the original parent bit I get, but do we really need this? Any context why a full grid change detection is needed/doing?

Reply:
`deleteRow` removes the record from the data, but the actual removal of the DOM element happens on the next change detection. This creates a bit of a timing issue. 

I no longer remember the details, since it has been a while, however I can confirm that the code does not work without the change detection (the issue reproduces without the change detection and is fixed with the change detection).